pkg/policy/validate.go

```diff
@@ -1066,9 +1066,10 @@ func podControllerAutoGenExclusion(policy kyvernov1.PolicyInterface) bool {
 func validateKinds(kinds []string, mock bool, client dclient.Interface, p kyvernov1.PolicyInterface) error {
 	for _, kind := range kinds {
 		gv, k := kubeutils.GetKindFromGVK(kind)
-		if k == p.GetKind() {
-			return fmt.Errorf("kind and match resource kind should not be the same")
-		}
+		// TODO: enable Kyverno Policy protection
```

Comment:
should this be uncommented?

Reply:
I commented out this block to enable Kyverno policy protection.
Since this block prevents verifying signature on Kyverno policy, we cannot list ClusterPolicy in match resource in Kyverno policy.
I would like to get your comment on how Kyverno policy protection should be enabled.
Let me remove "TODO" line because it is a bit confusing...

Reply:
@JimBugwadia do you see any original reason of preventing policy kind here in this block? We just want to confirm this to avoid unexpecetd side effect by comment out this part.

Reply:
Hi @yuji-watanabe-jp and @rurikudo - I am not sure and have asked on Slack here: 

https://kubernetes.slack.com/archives/C032MM2CH7X/p1660239251493439